wallet/controllers.go

```diff
@@ -0,0 +1,188 @@
+}
+
+// GetWalletResponse gets wallet info tied to a wallet id
+type GetWalletResponse struct {
+	Wallet *walletutils.Info `json:"wallet"`
+}
+
+// PostCreateWalletRequest has possible inputs for the new wallet
+type PostCreateWalletRequest struct {
+	Provider string `json:"provider" valid:"in(brave|uphold)"`
+	SignedTx string `json:"signedTx" valid:"-"`
+}
+
+// PostCreateWallet creates a wallet
+func PostCreateWallet(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		var req PostCreateWalletRequest
+		err := requestutils.ReadJSON(r.Body, &req)
+		if err != nil {
+			return handlers.WrapError(err, "Error unmarshalling body", http.StatusBadRequest)
+		}
+		_, err = govalidator.ValidateStruct(req)
+		if err != nil {
+			return handlers.WrapValidationError(err)
+		}
+
+		publicKey, err := middleware.GetKeyID(r.Context())
+		if err != nil {
+			return handlers.WrapError(err, "unable to look up http signature info", http.StatusBadRequest)
+		}
+
+		info, err := CreateWallet(req, publicKey)
+		if err != nil {
+			return handlers.WrapError(err, "unable to save wallet", http.StatusServiceUnavailable)
+		}
+		err = service.Datastore.InsertWallet(&info)
+		if err != nil {
+			return handlers.WrapError(err, "unable to save wallet", http.StatusServiceUnavailable)
+		}
+
+		return handlers.RenderContent(r.Context(), info, w, http.StatusCreated)
+	})
+}
+
+// GetWallet retrieves wallet information
+func GetWallet(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		paymentIDParam := chi.URLParam(r, "paymentId")
+		paymentID, err := uuid.FromString(paymentIDParam)
+
+		if err != nil {
+			return handlers.ValidationError("request url parameter", map[string]string{
+				"paymentId": "paymentId '" + paymentIDParam + "' is not supported",
+			})
+		}
+
+		info, err := service.Datastore.GetWallet(paymentID)
+		if err != nil {
+			return handlers.WrapError(err, "Error getting wallet", http.StatusNotFound)
+		}
+
+		// just doing this until another way to track
+		if info.AltCurrency == nil {
+			tmp := altcurrency.BAT
+			info.AltCurrency = &tmp
+		}
+
+		return handlers.RenderContent(r.Context(), info, w, http.StatusOK)
+	})
+}
+
+// CreateWallet creates a new set of wallet info
+func CreateWallet(req PostCreateWalletRequest, publicKey string) (walletutils.Info, error) {
+	provider := req.Provider // client
+
+	var info walletutils.Info
+	info.ID = uuid.NewV4().String()
+	info.Provider = provider
+	{
+		tmp := altcurrency.BAT
+		info.AltCurrency = &tmp
+	}
+
+	info.PublicKey = publicKey
+
+	if req.Provider == "uphold" {
+		if req.SignedTx != "" {
+			wallet := uphold.Wallet{
+				Info:    info,
+				PrivKey: ed25519.PrivateKey{},
+				PubKey:  httpsignature.Ed25519PubKey([]byte(publicKey)),
+			}
+			err := wallet.SubmitRegistration(req.SignedTx)
```

Comment:
I think we may be able to skip creating an Uphold card altogether

Reply:
as per dm, only allow uphold creation under local envs for testing purposes